tests/playwright/shiny/components/nav/test_nav.py

```diff
@@ -20,75 +20,60 @@ def test_nav(page: Page, local_app: ShinyAppProc) -> None:
     # Update the page size to be wider
     page.set_viewport_size({"width": 1500, "height": 800})
 
-    # make it in a funcn and then loop through it
-    # navset_tab
-    navset_tab = LayoutNavsetTab(page, "navset_tab")
-    navset_tab.expect_nav_values(["a", "b", "c"])
-    navset_tab.expect_value("a")
-    navset_tab.expect_content("navset_tab(): tab a content")
-    navset_tab.set("b")
-    navset_tab.expect_value("b")
-    navset_tab.expect_content("navset_tab(): tab b content")
+    nav_data = {
+        "navset_tab": {
+            "control": LayoutNavsetTab,
+            "verify": "navset_tab()"
+        },
+        "navset_pill": {
+            "control": LayoutNavSetPill,
+            "verify": "navset_pill()"
+        },
+        "navset_underline": {
+            "control": LayoutNavSetUnderline,
+            "verify": "navset_underline()"
+        },
+        "navset_card_tab": {
+            "control": LayoutNavSetCardTab,
+            "verify": "navset_card_tab()"
+        },
+        "navset_card_pill": {
+            "control": LayoutNavSetCardPill,
+            "verify": "navset_card_pill()"
+        },
+        "navset_card_underline": {
+            "control": LayoutNavSetCardUnderline,
+            "verify": "navset_card_underline()"
+        },
+        "navset_pill_list": {
+            "control": LayoutNavSetPillList,
+            "verify": "navset_pill_list()"
+        },
+        "page_navbar": {
+            "control": LayoutNavSetBar,
+            "verify": "page_navbar"
+        }
+    }
 
-    # # navset_pill
-    navset_pill = LayoutNavSetPill(page, "navset_pill")
-    navset_pill.expect_nav_values(["a", "b", "c"])
-    navset_pill.expect_value("a")
-    navset_pill.expect_content("navset_pill(): tab a content")
-    navset_pill.set("b")
-    navset_pill.expect_value("b")
-    navset_pill.expect_content("navset_pill(): tab b content")
+    def tester(el_name):
+        element = nav_data[el_name]['control'](page, el_name)
+        element.expect_nav_values(["a", "b", "c"])
+        element.expect_value("a")
+        element.expect_content(nav_data[el_name]['verify'] + ": tab a content")
+        element.set("b")
+        element.expect_value("b")
+        element.expect_content(nav_data[el_name]['verify'] + ": tab b content")
 
-    # navset_underline
-    navset_underline = LayoutNavSetUnderline(page, "navset_underline")
-    navset_underline.expect_nav_values(["a", "b", "c"])
-    navset_underline.expect_value("a")
-    navset_underline.expect_content("navset_underline(): tab a content")
-    navset_underline.set("b")
-    navset_underline.expect_value("b")
-    navset_underline.expect_content("navset_underline(): tab b content")
+    nav_list = [
+        "navset_tab", "navset_pill", "navset_underline", "navset_card_tab",
+        "navset_card_pill", "navset_card_underline", "navset_pill_list", "page_navbar"
+    ]
+    failures = []
+    # soft assert
+    for nav in nav_list:
```

Comment:
Use data already defined

```suggestion
    failures = []
    # soft assert
    for nav in list(nav_data.keys()):
```